### PR TITLE
[Frost DK] Might of the frozen wastes rank 1 + 2

### PR DESCRIPTION
**Pull request by Armadk**
Cherry picked two commits from bfa-dev, as it looks like they are still relevant, at least so far in SL.

### CODE REVIEW

File: engine/class_modules/sc_death_knight.cpp [hunk 1]
```diff
@@ -856,6 +859,7 @@ struct death_knight_t : public player_t {
   {
     double lucid_dreams_minor_proc_chance = 0.15;
     bool disable_aotd = false;
+    bool killing_machine_rppm = false;
```
Comment (Melekus) [inline engine/class_modules/sc_death_knight.cpp]: I don't think the rppm killing machine options and override are necessary going into shadowlands, it was mostly just a testing tool to compare how off it was compared to the new formula.

File: engine/class_modules/sc_death_knight.cpp [hunk 2]
```diff
@@ -6925,14 +6944,33 @@ unsigned death_knight_t::replenish_rune( unsigned n, gain_t* gain )
 }
 
 // Helper function to trigger Killing Machine, whether it's from a "forced" proc, a % chance, or the regular rppm proc
-void death_knight_t::trigger_killing_machine( double chance, proc_t* proc, proc_t* wasted_proc )
+void death_knight_t::trigger_killing_machine( double chance, proc_t* proc, proc_t* wasted_proc, weapon_e weapon_type )
```
Comment (Melekus) [inline engine/class_modules/sc_death_knight.cpp]: You don't need to pass the player's weapon type from the triggering actions to death_knight_t::trigger_killing_machine(), it should be accessible just fine within it.

File: engine/class_modules/sc_death_knight.cpp [hunk 3]
```diff
@@ -5394,7 +5404,15 @@ struct obliterate_strike_t : public death_knight_melee_attack_t
     background = special = true;
     may_miss = false;
     weapon = w;
-    base_multiplier *= 1.0 + p -> spec.obliterate_2 -> effectN( 1 ).percent();
+    if ( p -> spec.obliterate_2->ok() )
```
Comment (Melekus) [inline engine/class_modules/sc_death_knight.cpp]: This works just fine without the -> ok() check

File: engine/class_modules/sc_death_knight.cpp (hunk 1 above)
Comment (Armadk) [reply engine/class_modules/sc_death_knight.cpp]: Perfect, I'll pull those sections out.

**Pull request merged**